pkg/logs/tag/provider.go

```diff
@@ -20,22 +22,34 @@ type Provider interface {
 	GetTags() []string
 }
 
-// NoopProvider does nothing
-var NoopProvider Provider = &noopProvider{}
+var (
+	// NoopProvider does nothing
+	NoopProvider Provider = &noopProvider{}
+)
 
 // provider provides a list of up-to-date tags for a given entity by calling the tagger.
 type provider struct {
 	entityID             string
 	taggerWarmupDuration time.Duration
+	expectedTagsDuration time.Duration
+	submitExpectedTags   bool
 	sync.Once
+	sync.RWMutex
 }
 
 // NewProvider returns a new Provider.
 func NewProvider(entityID string) Provider {
-	return &provider{
+	p := &provider{
 		entityID:             entityID,
 		taggerWarmupDuration: config.TaggerWarmupDuration(),
 	}
+
+	if config.IsExpectedTagsSet() {
```

Comment:
I haven't seen it mentioned in the PR description, but I think the desired behavior is that when this option is enabled the host tags are sent for N minutes after _the agent startup_, whereas the current implementation sends host tags for N minutes after the _tag provider is created_.

These can be 2 very different things because tag providers are created dynamically for each log input (see for example in the [file tailer](https://github.com/DataDog/datadog-agent/blob/1e7c9a05211339eb44bd43015b1cc9333c57397a/pkg/logs/input/file/tailer.go#L91)), so, typically, with auto-discovered containers, a new tag provider is created for each container discovered.

I think this can be addressed with minimal code changes: the `config` package can keep track of the initialization time of the main config (which should be a good-enough approximation of the agent startup time), and add a method to `config` that returns the remaining duration during which the tag provider should send host tags.